doc/sphinx/source/iio_osc/data_streaming.rst

```diff
@@ -0,0 +1,127 @@
+   the IIO oscilloscope application must be upgraded to the most recent versions.
+
+Creating binary files for data streaming
+========================================
+
+:doc:`pyadi-iio: Python interfaces </source/python/python>`
+The PyADI module allows users to create binary data files from the input data using the
+tx class. The CLI (Command Line Interface) script adistream.py allows the user to 
+create various types of waveforms like sine, triangular, pwm, etc and store them to a 
+binary file.
+
+Loading and Streaming the Data
+==============================
+
+On the iio oscilloscope, open the DAC Data Manager tab as indicated below 
+after connecting to the iio device.
+
+   .. image:: /source/iio_osc/iio_osc_data_manager.jpg
+      :width: 600
+
+Click on the File Selection section to input the wavefrom data binary file created.
+Copy the binary file path and paste it in the Location section, then click Open.
+
+   .. image:: /source/iio_osc/iio_osc_file_section.jpg
+      :width: 600
+
+.. note::
+   
+   The input data file shall contain data according to the number of active channels 
+   enabled, and otherwise may result in incorrrect data being streamed.
+
+Once the required channels are enabled, use the cyclic buffer checkbox to enable 
+or disable cyclic streaming of data. Click the Load button to load the data. The 
+message "Waveform loaded successfully" shows up if the data is loaded correctly. 
+
+   .. image:: /source/iio_osc/iio_osc_load_data.jpg
+      :width: 600
+
+.. note::
+   
+   The Debug tab should not be accessed when streaming data, as this would 
+   impact data streaming. All uses same communication bus to access the data which 
+   could result into access/busy conflicts during data streaming.
+
+If Enable/Disable cyclic buffer is unchecked, streaming stops once all the data from
+the file is loaded, and if the cyclic buffer checkbox is checked, "Stop buffer transmission" 
+button needs to be clicked to end the data streaming.
+
+===================================================
+Limitations of Data Streaming Using IIO Application
+===================================================
+
+**DAC Sampling/Update Rate:**
+
+The sampling rate or update rate defines the maximum rate/speed at which data can 
+be loaded and updated onto the DAC output using the IIO firmware application. For 
+DAC's, typical time to update single sample is defined as:
+
+Time to update single sample: DAC data write time over digital interface + DAC conversion & output response time
+
+Data update rate is also limited because of additional overhead in the MCU firmware, 
+such as interrupt context switching time, SPI (digital interface) clock frequency, 
+MCU clock speed, etc.
+
+The "AD579X IIO Application" is used as a reference here. 
+For AD5790, which is a one-channel VDAC, the update rate is 71.4 KSPS. So, if a 
+sine wave of full scale with 200 data points is to be streamed, the output frequency 
```

Comment:
is 'full scale' significant here?

Reply:
not really on determining the f_out.. will remove it